webapp/src/main/java/com/box/l10n/mojito/okapi/filters/JSONFilter.java

```diff
@@ -94,6 +101,15 @@ public void handleValue(String value, JsonValueTypes valueType) {
         extractNoteIfMatch(value);
         extractUsageIfMatch(value);
         super.handleValue(value, valueType);
+        if (comment != null) {
+            addXliffNote(comment);
+            ITextUnit textUnit = getEventTextUnit();
+            if (textUnit != null) {
+                textUnit.setAnnotation(xliffNoteAnnotation);
+            }
+            comment = null;
```

Comment:
maybe put that logic in a function that says it handle "json comment" .

I suppose if you reset here it is because this is called after `handleObjectEnd`?

Reply:
Done.